### PR TITLE
Removed sseclient dependency + major enhancements on event_stream()

### PR DESCRIPTION
**Pull request by migueleliasweb**

### CODE REVIEW

File: marathon/client.py [hunk 1]
```diff
@@ -95,25 +95,23 @@ def _do_request(self, method, path, params=None, data=None):
...
-    def _do_sse_request(self, path, params=None, data=None):
-        from sseclient import SSEClient
-
-        headers = {'Accept': 'text/event-stream'}
-        messages = None
-        servers = list(self.servers)
-        while servers and messages is None:
-            server = servers.pop(0)
+    def _do_sse_request(self, path):
+        for server in list(self.servers):
```
Comment (Rob-Johnson) [inline marathon/client.py]: this is a change in behaviour. Previously, we were looping so long as we didn't have a response, but now we're doing the request on each server, irrespective of whether the request was successful. Please change it back :)

Comment (migueleliasweb) [inline marathon/client.py]: Okie dokie !

**Comment by solarkennedy** [comment]
Can you address the failing itests?
https://travis-ci.org/thefactory/marathon-python/jobs/154326294#L3594

**Comment by migueleliasweb** [comment]
Working on it !

File: marathon/client.py (hunk 1 above)
Comment (migueleliasweb) [inline marathon/client.py]: Sorry for the delay. I had so solve some personal problems on the last couple of days but I'm still going to fix this !

File: marathon/client.py [hunk 2]
```diff
@@ -95,25 +95,24 @@ def _do_request(self, method, path, params=None, data=None):
...
+                    stream=True,
+                    headers={'Accept': 'text/event-stream'},
+                    auth=self.auth
+                )
             except Exception as e:
                 marathon.log.error('Error while calling %s: %s', url, e.message)
 
-        if messages is None:
-            raise MarathonError('No remaining Marathon servers to try')
+            if response.ok:
```
Comment (migueleliasweb) [inline marathon/client.py]: @Rob-Johnson This should fix, right ?

**Comment by solarkennedy** [comment]
I do like the reduction in dependencies, but do we not need any SSE specific code that came with that client?

We at Yelp don't use this feature. I'm only slightly more hesitant to merge it.

@nuclon do you have any opinion?

**Comment by migueleliasweb** [comment]
@solarkennedy If you want I can leave the sseclient dependency for now, not a problem but I searched the codebase and only found use on the _do_sse_request.

See the previous code:

```python
def _do_sse_request(self, path, params=None, data=None):
        from sseclient import SSEClient

        headers = {'Accept': 'text/event-stream'}
        messages = None
        servers = list(self.servers)
        while servers and messages is None:
            server = servers.pop(0)
            url = ''.join([server.rstrip('/'), path])
            try:
                messages = SSEClient(url, params=params, data=data, headers=headers,
                                     auth=self.auth)
            except Exception as e:
                marathon.log.error('Error while calling %s: %s', url, e.message)

        if messages is None:
            raise MarathonError('No remaining Marathon servers to try')

        return messages
```

The **sseclient** is only imported on this very method.

I'm not pushing but I think this is kinda safe.

Feel free to ask any question.

**Comment by nuclon** [comment]
@solarkennedy I've made small pull request to sseclient recently and it fixes that weird CPU problem.
https://bitbucket.org/btubbs/sseclient/pull-requests/9/response-readline/diff
and now everything works well. (we're using this feature in our company)

Of course it's your call to decide - remove the dependency or not.

**Comment by migueleliasweb** [comment]
@nuclon Instead of iterating char by char you made the iteration by line, am I right ? If so, that should do.

```diff
def __next__(self):
         while not self._event_complete():
             try:
-                nextchar = next(self.resp_iterator)
-                self.buf += nextchar
-            except (StopIteration, requests.RequestException):
+                nextline = self.resp_file.readline()
+                if not nextline:
+                    raise EOFError()
+                self.buf += nextline
+            except (StopIteration, requests.RequestException, EOFError) as e:
                 time.sleep(self.retry / 1000.0)
                 self._connect()
```

I would like (as a bonus of solving this problem) removing an external dependency but the truth is both  codes would solve the issue with huge json responses from marathon.

@solarkennedy Feel free to chose any. I just like to solve this problem asap. I have a running code on production that might snap at any time =/

File: marathon/client.py (hunk 2 above)
Comment (Rob-Johnson) [inline marathon/client.py]: yeah, this is fine.

**Comment by solarkennedy** [comment]
I agree that removing the dep is good too. I'll make a new release to include this change.

**Comment by migueleliasweb** [comment]
Hooray !

**Comment by dmsimard** [comment]
seeclient is still marked in setup.py as a dependency: https://github.com/thefactory/marathon-python/blob/master/setup.py#L16

**Comment by migueleliasweb** [comment]
I saw that after the merge was made. I ll make another pr to remove it from
the setup.py.

**Pull request merged**